### PR TITLE
[Tests-Only] Added acceptance test for changing existing expiration date on collaborator share

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
Acceptance test added for changing existing expiration date of an existing user/group share

## Related Issue
- Part of #3132

## How Has This Been Tested?

- locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js [hunk 1]
```diff
@@ -485,6 +485,30 @@ module.exports = {
       }
 
       return this.useXpath().expect.element(collaboratorSelector).to.not.be.present
+    },
+    /**
+     * @param key
+     * @param value
+     * @return {Promise<module.exports.commands>}
+     */
+    setCollaboratorsForm: async function (key, value) {
```
Comment (dpakach) [inline tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: If this function just accepts one key `expireDate` why even check it. Can we just call it `setExpireDate()`?

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 2]
```diff
@@ -128,6 +128,41 @@ const shareFileFolder = function (
...
+ * create any share using dataTable
+ *
+ * @param {string} sharer
+ * @param {object} dataTable
+ * @param {string} dataTable.path
+ * @param {string} dataTable.shareWith
+ * @param {string} dataTable.expireDate
+ * @param {string} dataTable.name
+ * @param {string} dataTable.permissionString
+ * @param {string} dataTable.password
```
Comment (jasson99) [inline tests/acceptance/stepDefinitions/sharingContext.js]: I think these can be skipped.

File: tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js (hunk 1 above)
Comment (kiranparajuli589) [reply tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: yes, ok! :+1:

File: tests/acceptance/stepDefinitions/sharingContext.js (hunk 2 above)
Comment (kiranparajuli589) [reply tests/acceptance/stepDefinitions/sharingContext.js]: i've changed the doc.
hera again!!

File: tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js [hunk 3]
```diff
@@ -190,7 +190,7 @@ module.exports = {
         .waitForOutstandingAjaxCalls()
         .waitForElementNotPresent('@addShareSaveButton')
     },
-    saveCollaboratorPermission: function () {
+    saveCollaboratorEditForm: function () {
```
Comment (HariBhandari07) [inline tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: I think a better name would be `saveChanges`

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 4]
```diff
@@ -128,6 +128,36 @@ const shareFileFolder = function (
...
+ * @param {object} dataTable (attrs like: path, shareWith, expireDate, name, permissionString,
+ * password can be passed inside dataTable)
+ *
+ * @return void
+ */
+Given('user {string} has created a new share with following settings',
+  function (sharer, dataTable) {
+    const settings = dataTable.rowsHash()
+    let expireDate = settings.expireDate
+    if (typeof settings.expireDate !== 'undefined') {
```
Comment (HariBhandari07) [inline tests/acceptance/stepDefinitions/sharingContext.js]: here you can use `expireDate` instead of `settings.expireDate`

File: tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js (hunk 3 above)
Comment (kiranparajuli589) [reply tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: done! :+1:

File: tests/acceptance/stepDefinitions/sharingContext.js (hunk 4 above)
Comment (kiranparajuli589) [reply tests/acceptance/stepDefinitions/sharingContext.js]: done! :rocket:

File: tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js [hunk 5]
```diff
@@ -485,6 +485,20 @@ module.exports = {
...
+     * @param value
+     *
+     * @return {Promise<*>}
+     */
+    changeCollaboratorExpiryDate: async function (collaborator, value) {
+      await collaboratorDialog.clickEditShare(collaborator)
+      await this.api.page
+        .FilesPageElement
+        .publicLinksDialog()
+        .setPublicLinkExpiryDate(value)
```
Comment (individual-it) [inline tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: now we are using that pageObject from the sharingDialog, so its not really part of the publicLinksDialog, we need to move it

Comment (kiranparajuli589) [reply tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: yes, @individual-it, I've done this in PR #3165!

Comment (individual-it) [reply tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: I've reviewed that PR

Comment (kiranparajuli589) [reply tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js]: @individual-it you can review again!

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 6]
```diff
@@ -87,7 +87,7 @@ Given('user {string} from remote server has shared {string} with user {string} f
  * @param {string} elementToShare  path of file/folder being shared
  * @param {string} sharer  username of the sharer
  * @param receiver  username of the receiver
- * @param shareType  Type of share 0 = user, 1 = group, 3 = public (link), 6 = federated (cloud share).
+ * @param {number} shareType  type of share user, group, public (link), federated (cloud share).
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/sharingContext.js]: why changing that comment? how is this change relevant to this PR?

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 7]
```diff
@@ -742,6 +777,17 @@ When(
   }
 )
 
+When('the user edits share with user/group {string} of file/folder/resource {string} changing following',
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/sharingContext.js]: ìn phoenix its called "collaborator" not "share" and  please add "using the webUI" to the sentence

Comment (kiranparajuli589) [reply tests/acceptance/stepDefinitions/sharingContext.js]: done!

File: tests/acceptance/stepDefinitions/sharingContext.js (hunk 6 above)
Comment (kiranparajuli589) [reply tests/acceptance/stepDefinitions/sharingContext.js]: My bad, I thought this would be revealing information
reverted back to as it is!
